lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -87,16 +86,10 @@ export async function generateLockFile(
 
     let args = '--lockfile-only';
 
-    // pnpm v9+ is doing recursive automatically when it detects workspaces.
-    //
     // If it's not a workspaces project/monorepo, but single project with unrelated other npm project in source tree (for example, a git submodule),
     // `--recursive` will install un-wanted project.
     // we should avoid this.
-    if (
-      pnpmToolConstraint.constraint &&
-      !semver.intersects(pnpmToolConstraint.constraint, '>=9') &&
-      (await localPathExists(pnpmWorkspaceFilePath))
-    ) {
+    if (await localPathExists(pnpmWorkspaceFilePath)) {
```

Comment:
So this will now mean that we'll add the --recursive flag to pnpm every time the workspace file exists?

Reply:
seems to be wrong because the file now can also only contain pnpm settings. we should only add the recursive if it contains the packages key. I think we've an open PR about it.